packages/fmg-nitro-adjudicator/test/consensus-app.test.ts

```diff
@@ -65,197 +39,281 @@ describe('ConsensusApp', () => {
     '6370fd033278c143179d81c5526140625662b8daa446c22ee2d73db3707e620c',
   );
   const participantC = new ethers.Wallet(
-    '5e1b32fb763f62e1d19a9c9cd8c5417bd31b7d697ee018a8afe3cac2292fdd3e'
+    '5e1b32fb763f62e1d19a9c9cd8c5417bd31b7d697ee018a8afe3cac2292fdd3e',
   );
   const participants = [participantA.address, participantB.address, participantC.address];
-  const NUM_PARTICIPANTS = participants.length;
   const proposedDestination = [participantA.address, participantB.address];
 
   const allocation = [toUint256(1), toUint256(2), toUint256(3)];
   const proposedAllocation = [toUint256(4), toUint256(2)];
 
   const channel: Channel = { channelType: participantB.address, nonce: 0, participants }; // just use any valid address
-  const defaults = { channel, allocation, destination: participants, turnNum: 6, proposedDestination, proposedAllocation, commitmentCount: 0 };
+  const defaults = {
+    channel,
+    allocation,
+    destination: participants,
+    turnNum: 6,
+    proposedDestination,
+    proposedAllocation,
+    commitmentCount: 0,
+  };
+
+  const oneVoteComplete = propose(
+    initialConsensus(defaults),
+    proposedAllocation,
+    proposedDestination,
+  );
+  const twoVotesComplete = vote(
+    propose(initialConsensus(defaults), proposedAllocation, proposedDestination),
+  );
 
   beforeAll(async () => {
     await setupContracts();
   });
 
-  it.skip('reverts when the proposed allocation and the proposed destination are the incorrect length', async () => {
-    // I think this test should fail, since this requirement prevents
-    // the nitro protocol
-    const fromCommitment = commitments.appCommitment({
-      ...defaults,
-      turnNum: 6,
-      consensusCounter: 0,
+  describe('the propose transition', async () => {
+    const fromCommitment = initialConsensus(defaults);
+    const toCommitment = propose(fromCommitment, proposedAllocation, proposedDestination);
+    it('returns true on a valid transition', async () => {
+      await validTransition(fromCommitment, toCommitment);
     });
-    const toCommitment = commitments.appCommitment({
-      ...defaults,
-      allocation: proposedAllocation,
-      destination: participants,
-      turnNum: 6,
-      consensusCounter: 1,
-      proposedAllocation: [toUint256(1), toUint256(2)],
-      proposedDestination: [participantA.address],
-    });
-    invalidTransition(fromCommitment, toCommitment, "ConsensusApp: newCommitment.proposedAllocation.length must match newCommitment.proposedDestination.length");
+
+    itRevertsWhenTheBalancesAreChanged(fromCommitment, toCommitment);
```

Comment:
These functions are `async` functions. I thought you would need an `await` to ensure that the assertions are actually run.

Reply:
The test helper functions aren't actually async, they just contain async `it` statements, so I don't think we need to `await` them.

Reply:
Ah, right, that makes sense.

It would be a problem if the helper functions contained async expectations, and were run within an async `it` callback.